.github/workflows/checks.yml

```
@@ -7,27 +7,65 @@ on:
   pull_request:
     types: [opened, synchronize, reopened]
   workflow_dispatch:
+    inputs:
+      force:
+        description: 'Force running all checks'
+        required: false
+        default: 'false'
 
 jobs:
-  cargo-check:
-    name: Cargo Check
+  files-changed:
+    name: Detect what files changed
     runs-on: [self-hosted]
+    outputs:
+      markdown: ${{ steps.filter.outputs.markdown }}
+      rust: ${{ steps.filter.outputs.rust }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Detect changed file types
+        uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            markdown:
+              - .markdownlint.yaml
+              - '**/*.md'
+            rust:
+              - '**/*.lock'
+              - '**/*.rs'
+              - '**/*.toml'
```

I suggest checking all `include!` and `include_bytes!` macros to define all of the Rust code dependencies.

json files could be changed here and it will not be covered in CI:
https://github.com/QuantumFusion-network/qf-solochain/blob/khssnv/skip-irrelevant-checks/node/src/chain_spec.rs#L166

The Makefile could be changed and it will not be checked in CI:
https://github.com/QuantumFusion-network/qf-solochain/blob/482e90b708ee2b8c6a6cebf0c400d126b28b75d8/.github/workflows/checks.yml#L71

```suggestion
            rust:
              - '**/*.json'
              - '**/*.lock'
              - '**/*.rs'
              - '**/*.toml'
              - 'Makefile'
```

---

Thanks, added. JSON tracking is added only for `node/src/res`.